### PR TITLE
feat: wrap

### PR DESCRIPTION
**Pull request by gavmck**
# Wrap component with no JS.
Creates typings, but adds the CSS to the global CSS.

This is quite nifty. Has the benefits of typescript typings and a web component interface with lightening fast loading due to the global CSS. We use PostCSS to import the CSS and cause it to be output into the head of the document when the component is loaded.

### CODE REVIEW

File: components/composition/Wrap/Wrap.css [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/* stylelint-disable selector-no-qualifying-type */
```
Comment (DanWebb) [inline components/composition/Wrap/Wrap.css:1]: I think we can use nesting to not have to repeat diamond-wrap? https://developer.mozilla.org/en-US/docs/Web/CSS/Nesting_selector

Comment (DanWebb) [inline components/composition/Wrap/Wrap.css:1]: We should set this rule to ignore attribute https://stylelint.io/user-guide/rules/selector-no-qualifying-type/#ignore-attribute-class-id

File: components/composition/Wrap/Wrap.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export {};
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'diamond-nojs-wrap': {
```
Comment (DanWebb) [inline components/composition/Wrap/Wrap.ts:5]:
```suggestion
    'diamond-wrap': {
```

File: example/index.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
   <script type="module" src="/index.js"></script>
 </head>
   <body>
-    <diamond-container>
+    <diamond-wrap size="tablet" gutter="md">
```
Comment (DanWebb) [inline example/index.html:11]: `size="tablet"` is a bit strange, it's the size of a breakpoint and the breakpoints aren't exposed.

Should we have more standard sizes (sm, md, lg) instead then have the wrap sizes in a CSS variable so they can be optionally overridden?

File: styles/diamond-ui.css [hunk 4]
```diff
@@ -1,3 +1,5 @@
 @import url('../node_modules/modern-normalize/modern-normalize.css');
 @import url('./tokens.css');
 @import url('./base.css');
+
+@import url('../components/composition/Wrap/Wrap.css');
```
Comment (DanWebb) [inline styles/diamond-ui.css:5]: We could add https://www.npmjs.com/package/postcss-import-ext-glob to pull these in, doesn't necessarily need to be part of this PR though.

**Review by DanWebb — CHANGES_REQUESTED** [review]

File: components/composition/Wrap/Wrap.css (hunk 1 above)
Comment (gavmck) [reply components/composition/Wrap/Wrap.css:1]: Not sure CSS nesting is ready for prod yet? https://caniuse.com/css-nesting

File: styles/diamond-ui.css (hunk 4 above)
Comment (gavmck) [reply styles/diamond-ui.css:5]: Added a ticket for this one

File: example/index.html (hunk 3 above)
Comment (gavmck) [reply example/index.html:11]: Yeah, that's probably more flexible

File: components/composition/Wrap/Wrap.css (hunk 1 above)
Comment (gavmck) [reply components/composition/Wrap/Wrap.css]: We could polyfill it via postCSS, I was kinda trying to stay away from a processor until I had to

Comment (DanWebb) [reply components/composition/Wrap/Wrap.css]: It looks fine on can i use it can be processed by consuming sites anyways

**Review by DanWebb — APPROVED** [review]

File: components/composition/Wrap/Wrap.css (hunk 1 above)
Comment (gavmck) [reply components/composition/Wrap/Wrap.css]: No support on Samsung internet at 3.36% of UK users is the one that stands out to me

Comment (gavmck) [reply components/composition/Wrap/Wrap.css]: But if it's preprocessable, maybe we should just do it

**Pull request merged**